### PR TITLE
cedar: init at 4.2.2

### PR DESCRIPTION
**Pull request by meain**
https://github.com/cedar-policy/cedar
Implementation of the Cedar Policy Language (Includes the rust cli to run, validate, format etc)

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ce/cedar/package.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  src = fetchFromGitHub {
+    owner = "cedar-policy";
+    repo = "cedar";
+    rev = "v${version}";
+    hash = "sha256-p8idQx3HGO/ikL0pDTPXx5+rD2sRpXANqs/g51BL554=";
+  };
+
+  cargoHash = "sha256-GuXifjQkH8jV9mT3UYU6rUZB1qO5Xl9tvYJW7MjcW0U=";
+
```
Comment (getchoo) [inline pkgs/by-name/ce/cedar/package.nix:19]:
```suggestion

  passthru = {
    tests.version = testers.testVersion { package = cedar; };
    updateScript = nix-update-script { };
  };

```

Not required, but I'd recommend adding a basic test and update script to make this a bit more maintainable

File: pkgs/by-name/ce/cedar/package.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "cedar";
+  version = "4.2.2";
+
+  src = fetchFromGitHub {
+    owner = "cedar-policy";
+    repo = "cedar";
+    rev = "v${version}";
```
Comment (getchoo) [inline pkgs/by-name/ce/cedar/package.nix:14]:
```suggestion
    rev = "refs/tags/v${version}";
```
Nitpick

File: pkgs/by-name/ce/cedar/package.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    hash = "sha256-p8idQx3HGO/ikL0pDTPXx5+rD2sRpXANqs/g51BL554=";
+  };
+
+  cargoHash = "sha256-GuXifjQkH8jV9mT3UYU6rUZB1qO5Xl9tvYJW7MjcW0U=";
+
+  meta = {
+    description = "Implementation of the Cedar Policy Language";
+    homepage = "https://github.com/cedar-policy/cedar";
```
Comment (getchoo) [inline pkgs/by-name/ce/cedar/package.nix:22]:
```suggestion
    homepage = "https://github.com/cedar-policy/cedar";
    changelog = "https://github.com/cedar-policy/cedar/releases/tag/v${version}";
```
Linking to the changelog would also be nice

**Review by getchoo — APPROVED** [review]
Some notes if you have the time. None are blocking though, and LGTM

**Comment by meain** [comment]
@getchoo really appreciate the review. I was not aware of the testing pattern.

**Comment by getchoo** [comment]
You will also want to format this with `nixfmt`: https://github.com/NixOS/nixpkgs/actions/runs/12102372743/job/33743368686?pr=360328

**Review by getchoo — CHANGES_REQUESTED** [review]
Almost forgot: you will also want to format this with nixfmt. See https://github.com/NixOS/nixpkgs/actions/runs/12102372743/job/33743368686?pr=360328

**Comment by getchoo** [comment]
Yup :(

Lets try it again

**Comment by getchoo** [comment]
@ofborg eval

**Comment by meain** [comment]
I did a rebase and force push to rerun the CI. The eval command to ofborg did not seem to do anything somehow.

**Review by getchoo — APPROVED** [review]

**Comment by getchoo** [comment]
Feel free to ping me in a couple days if no one else merges :)

**Comment by meain** [comment]
> Feel free to ping me in a couple days if no one else merges :)

@getchoo ✋🏼

**Pull request merged**